File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260917-061856-52134/analysis/own-tests-project/src/main/java/surveycrawler/FirmSand.java
package surveycrawler;

public class FirmSand extends Ground {

    // moving

    @Override
    public void moveCrawlerTowards(SurveyCrawler aCrawler, Point aDirection) {
        aCrawler.displaceBy(aDirection);
    }

    // facing

    @Override
    public void turnCrawler(SurveyCrawler aCrawler, Runnable aTurn) {
        aCrawler.turnDoing(aTurn);
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-204513-77784/workdir/src/surveycrawler/firm_sand.py
from surveycrawler.ground import Ground


class FirmSand(Ground):

    # moving

    def move_crawler_towards(self, a_crawler, a_direction):
        a_crawler.move_one_cell_towards(a_direction)

    # facing

    def assert_crawler_can_turn(self, a_crawler):
        pass
